cur-lib/cur/stdlib/sugar.rkt

```diff
@@ -184,53 +187,86 @@
 ;; - but other tys are named
 ;; - in each clause, there must be one pat per ty-to-match,
 ;;   where the pats are stx-parse pats
-;; TODO:
-;; - check smaller arg for rec calls
-;; - check coverage of pats
+
+;; TODO PR103:
 ;; - check that body has type ty_out; currently, mismatch wont error
-;;(require (for-syntax racket/pretty))
+;; (this may have been added in PR103)
+
+;; TODO PR103: Since define/rec/match and define/rec/match^ match the same syntax,
+;; should abstract the syntax-parser.
 (define-syntax (define/rec/match stx)
   (syntax-parse stx
     [(_ name:id
         (~datum :)
-        [x (~datum :) ty_in1] ...
+        (~and decls-x [x (~datum :) ty_in1]) ...
         (~and ty-to-match (~not [_ (~datum :) _]) (~not (~datum ->))) ...
-        [y (~datum :) ty_in2] ...
+        (~and decls-y [y (~datum :) ty_in2]) ...
         (~datum ->) ty_out
-        [pat ... (~datum =>) body] ...)
-     #:with (decls-x ...) (for/list ([x (attribute x)]
-                                     [t (attribute ty_in1)])
-                            #`(#,x : #,t))
-     #:with (decls-y ...) (for/list ([y (attribute y)]
-                                     [t (attribute ty_in2)])
-                            #`(#,y : #,t))
-     #:with (pat-bodies ...) (for/list ([pats (attribute pat)]
-                                        [body (attribute body)])
-                               #`(#,pats => #,body))
+        (~and clauses
+              [pat ... (~datum =>) body]) ...
+        (~optional
+         (~seq #:type-aliases
+               ([ty-in-aliases (~datum =) ty-out-aliases
+                               (~optional
+                                ty-out-implicit-counts:exact-nonnegative-integer
+                                ;; TODO PR103: Why is this a syntax-object, while others
```

Comment:
I used this because I wanted a default argument, but as was the case with me not knowing that `~and` was a thing, I'm sure things can be cleaned up quite a bit here